tests/Google/Service/ResourceTest.php

```diff
@@ -193,4 +197,93 @@ public function testAppEngineSslCerts()
         $this->client->getHttpClient()->getDefaultOption('verify')
     );
   }
+
+  public function testNoExpectedClassForAltMediaWithHttpSuccess()
+  {
+    // set the "alt" parameter to "media"
+    $request = new Request('GET', '/?alt=media');
+    $body = Stream::factory('thisisnotvalidjson');
+    $response = new Response(200, [], $body);
+    $arguments = [[]];
+
+    $http = $this->getMockBuilder("GuzzleHttp\Client")
+        ->disableOriginalConstructor()
+        ->getMock();
+    $http->expects($this->once())
+        ->method('send')
+        ->will($this->returnValue($response));
+    $http->expects($this->any())
+        ->method('createRequest')
+        ->will($this->returnValue($request));
+    $client = new Google_Client();
+    $client->setHttpClient($http);
+    $service = new Test_Google_Service($client);
+
+    // set up mock objects
+    $resource = new Google_Service_Resource(
+      $service,
+      "test",
+      "testResource",
+      array("methods" =>
+        array(
+          "testMethod" => array(
+            "parameters" => array(),
+            "path" => "method/path",
+            "httpMethod" => "POST",
+          )
+        )
+      )
+    );
+
+    $decoded = $resource->call('testMethod', $arguments);
```

Comment:
Should pass in an expected class here. One of the key things is ignoring the expected class when alt=media is present. Not specifying one in the first place can mask regressions.